src/utils.rs

```diff
@@ -6,3 +9,18 @@ pub fn set_threading(num_threads: usize) {
         .build_global()
         .unwrap();
 }
+
+cached_key! {
+    READ: SizedCache<String, Array<f64, IxDyn>> = SizedCache::with_size(10);
+    Key = { format!("{}", path)};
+    fn read_nd_image(path: String) -> Array<f64, IxDyn> = {
```

Comment:
Right, using dynamic dimension is clearly a good idea with this project. It adds a new type of bugs though, like 3D + 4D. Is it currently useful, even without any "axis" operations, like sum, mean, etc.?

Reply:
If the user try to do a 3D + 4D for now it will fail with an error like:
```
thread 'main' panicked at 'assertion failed: part.equal_dim(&self.dimension)'
```
This check is done on NDarray part. For now I don't want to provide 3D with 4D operations.